lib/spack/spack/abi.py

```diff
@@ -82,9 +82,9 @@ def _intel_compiler_compare(self, pversion, cversion):
         are ABI compatible"""
 
         # Test major and minor versions.  Ignore build version.
-        if len(pversion.version) < 2 or len(cversion.version) < 2:
-            return False
-        return pversion.version[:2] == cversion.version[:2]
+        pv = pversion.lo
+        cv = cversion.lo
```

Comment:
Do we always get `ClosedOpenRange`s from the calling function?

Reply:
Yes

Reply:
@becker33 Can we add type-hints to `abi.py` to have a check on that? Otherwise this LGTM

Reply:
FWIW.  I tried to help by adding type hints to this method BUT I don't seem to be able to push the changes to this PR.